TypeScript (#1819)
enkidevs/curriculum: typescript/introduction/intro-typescript/typescript-dependencies.md

---
author: kapnobatai136

type: normal

category: must-know

aspects:
  - introduction

---

# TypeScript Dependencies

---
## Content

Installing TypeScript is as easy as writing a line of code. Using Node's package manager, all you'd have to type is:

```bash
npm install -g typescript
```

> The above command will install TypeScript as a global package and you can start using it anywhere. Whenever you write TypeScript code, this is have to go in a `.ts` file.

Reviewer: ```suggestion
The above command will install TypeScript as a global package and you can start using it anywhere. A file containing TypeScript code ends in `.ts`.
```

```ts
// enki.ts
let name = 'Enki';

console.log(`${name} is cool!`);
```

> Note: this is plain JS code; in the following insight we will take a look at some TS code.

Reviewer: If above comment is merged, this one can be removed

> Because TypeScript is just a wrapper, all the code needs to be compiled to JavaScript:

Reviewer: ```suggestion
The conversion back from TypeScript to JavaScript has an extra conversion (called "compilation") step. To get JavaScript, we first have to remove any TypeScript-specific code that isn't valid in JavaScript. This compilation is done with the TypeScript Compiler (called `tsc`): 
```

```bash
tsc enki.ts
```

The result will be:

```js
// enki.js
let name = 'Enki';

console.log(`${name} is cool!`);
```

---
## Practice

> TypeScript code is always compiled to ???.
> 
> * JavaScript
> * Python
> * Angular
> * Go

Reviewer: ```suggestion
The technical term for converting TypeScript code into JavaScript is ???.

* compilation
* conversion
* transformation
* manipulation
```

---
author: kapnobatai136

type: normal

category: must-know

aspects:
  - introduction

---

# Get started with TypeScript

---
## Content

> Installing TypeScript is as easy as writing a line of code. Using Node's package manager, all you have to type is:

Reviewer 2: I'd drop `Node's package manager` and use `npm` directly

Reviewer 3: 👍 since there are so many expansions https://github.com/npm/npm-expansions

Reviewer: Yeah, good point 👍

Reviewer: ```suggestion
Installing TypeScript is as easy as writing a line of code. Using NPM, all you have to type is:
```

```bash
npm install -g typescript
```

The command above will install TypeScript as a global package and you can start using it anywhere. A file containing TypeScript code ends in `.ts`.

In fact, since any valid JavaScript is also TypeScript (types are optional), to convert a JavaScript file to TypeScript, we can just replace `.js` with `.ts`[1]:

```ts
// enki.ts
let name = 'Enki';

console.log(`${name} is cool!`);
```

The conversion back from TypeScript to JavaScript has an extra step (called "compilation"). To get JavaScript, we first have to remove any TypeScript-specific code that isn't valid in JavaScript. This step is done with the TypeScript Compiler (called `tsc`):

```bash
tsc enki.ts
```

This will compile a TypeScript file `enki.ts` into a JavaScript file `enki.js`:

```js
// enki.js
let name = 'Enki';

console.log(`${name} is cool!`);
```

---
## Practice

The technical term for converting TypeScript code into JavaScript is ???.

* compilation
* conversion
* transformation
* manipulation

---
## Revision

To compile a TypeScript file, you would write:

```bash
??? file.ts
```

* tsc
* ts
* jsc
* typescript

---
## Footnotes

> [1:Interoperability]

Reviewer: ```suggestion
[1:Interoperability with JavaScript]
```

> This is a very important feature of TypeScript. In fact, the #1 reason for its popularity is probably this interoperability with JavaScript and the fact that it's this easy to get started.

Reviewer: ```suggestion
This is a very important feature of TypeScript. In fact, a major reason for TypeScript's popularity is how easy it is for JavaScript developers to start using it.
```
